src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -88,12 +90,42 @@ const BelowTheFoldArtworkDetails: React.FC<BelowTheFoldArtworkDetailsProps> = ({
 
 export const ArtworkApp: React.FC<Props> = props => {
   const { artwork, me, referrer, tracking, shouldTrackPageView } = props
-  const { match, silentPush } = useRouter()
+  const { match, silentPush, silentReplace } = useRouter()
+  const { showAuthDialog } = useAuthDialog()
 
   const privateArtworksEnabled = useFeatureFlag(
     "amber_artwork_visibility_unlisted"
   )
 
+  useEffect(() => {
```

Comment:
Can you leave a comment on what exactly this is doing and why? It's really unclear just reading the code.